### PR TITLE
Some Rust cleanups

### PR DESCRIPTION
**Pull request by BartMassey**
Cleaned up a few things, for informational purposes.

### CODE REVIEW

File: calc.rs [hunk 1]
```diff
@@ -66,20 +66,19 @@ fn lex(c: &mut char) -> (Token, Option<f64>) {
...
             '+' => TPlus,
             '-' => TMinus,
             '*' => TTimes,
             '/' => TDivide,
             '(' => TOp,
             ')' => TCp,
+            c0 if c0.is_ascii_digit() => loop {
+                val = val * 10.0 + (*c as u32 - '0' as u32) as f64;
+                *c = getc();
+                if c.is_ascii_digit() {
```
Comment (keith-packard) [inline calc.rs]: Should be !c.is_ascii_digit?

**Review by keith-packard — COMMENTED** [review]
Thanks for taking a look. I'm going to explore Option types to see if can clean up token stuff.

File: calc.rs (hunk 1 above)
Comment (BartMassey) [reply calc.rs]: Yeah, did that at the last minute. Apologies.

**Comment by BartMassey** [comment]
What I would probably do for token types depends on how fancy I wanted to get. At minimum, I'd probably replace `Number` with `Number(f64)`. This will be a pain though: I started to do it and gave up for lack of time.

**Comment by keith-packard** [comment]
> What I would probably do for token types depends on how fancy I wanted to get. At minimum, I'd probably replace `Number` with `Number(f64)`. This will be a pain though: I started to do it and gave up for lack of time.

Already done.

**Comment by BartMassey** [comment]
Looks like this isn't worth merging at this point. Too diverged from current main

**Comment by BartMassey** [comment]
Do recommend using `cargo clippy` and `cargo fmt` liberally, though.

**Comment by keith-packard** [comment]
yeah, I pulled in the bits  I liked and dropped the rest on the floor.

**Pull request closed without merging**